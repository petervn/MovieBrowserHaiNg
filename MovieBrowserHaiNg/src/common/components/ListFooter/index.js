import React from 'react';
import {View, Text, TouchableOpacity, ActivityIndicator} from 'react-native';
import styles from './style';
export default ListFooter = props => {
  const {loadMore, isLoading, totalPages, page} = props;
  return (
    <View style={styles.loadingMore}>
      {isLoading && <ActivityIndicator color="#00ff00" size="large" />}
      {totalPages !== page && (
        <TouchableOpacity style={styles.loadingButton} onPress={loadMore}>
          <Text style={styles.loadingText}>Load more</Text>
        </TouchableOpacity>
      )}
    </View>
  );
};
