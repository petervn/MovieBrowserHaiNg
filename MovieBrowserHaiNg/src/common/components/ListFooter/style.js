import {StyleSheet} from 'react-native';

import {lightGray, darkBlue} from '../../styles/color';

const styles = StyleSheet.create({
  loadingMore: {
    marginBottom: 30,
    paddingBottom: 20,
    paddingBottom: 50,
    justifyContent: 'center',
    alignItems: 'center',
  },
  loadingButton: {
    padding: 10,
    width: '50%',
    borderWidth: 1,
    borderRadius: 100,
    borderColor: lightGray,
    backgroundColor: 'purple',
  },
  loadingText: {
    fontSize: 12,
    color: 'white',
    textAlign: 'center',
  },
});

export default styles;
