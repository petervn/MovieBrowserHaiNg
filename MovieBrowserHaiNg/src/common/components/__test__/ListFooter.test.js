import React from 'react';
import {Text, ActivityIndicator} from 'react-native';
import {shallow, mount} from 'enzyme';
import {expect as chaiExpect} from 'chai';
import ListFooter from '../ListFooter';

describe('List Footer', () => {
  let loadMore = () => {},
    isLoading = true,
    totalPages = 10,
    page = 1;

  describe('Rendering', () => {
    it('should match to snapshot', () => {
      const component = shallow(
        <ListFooter
          isLoading={isLoading}
          totalPages={totalPages}
          loadMore={loadMore}
          page={page}
        />,
      );
      expect(component).toMatchSnapshot();
    });

    it('should show ActivityIndicator If isLoading is true', () => {
      let component = mount(
        <ListFooter
          isLoading={isLoading}
          totalPages={totalPages}
          loadMore={loadMore}
          page={page}
        />,
      );
      const activityIndicator = component.find(ActivityIndicator);
      chaiExpect(activityIndicator).to.have.lengthOf(1);
      // use mount() insted of using shallow to invoke .text()
      const textValue = component.find(Text).text();
      expect(textValue).toEqual('Load more');
    });

    it('should hide ActivityIndicator If isLoading is flase', () => {
      let component = mount(
        <ListFooter
          isLoading={false}
          totalPages={totalPages}
          page={page}
          loadMore={loadMore}
        />,
      );
      const activityIndicator = component.find(ActivityIndicator);
      chaiExpect(activityIndicator).to.have.lengthOf(0);
    });

    it('should hide Loadmore button If current page is the last page', () => {
      let component = mount(
        <ListFooter
          isLoading={false}
          totalPages={10}
          page={10}
          loadMore={loadMore}
        />,
      );
      chaiExpect(Text).to.have.lengthOf(0);
    });
  });
});
