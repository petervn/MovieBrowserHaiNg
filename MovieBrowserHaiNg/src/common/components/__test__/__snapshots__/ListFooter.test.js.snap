// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List Footer Rendering should match to snapshot 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "justifyContent": "center",
      "marginBottom": 30,
      "paddingBottom": 50,
    }
  }
>
  <ActivityIndicator
    animating={true}
    color="#00ff00"
    hidesWhenStopped={true}
    size="large"
  />
  <TouchableOpacity
    activeOpacity={0.2}
    onPress={[Function]}
    style={
      Object {
        "backgroundColor": "purple",
        "borderColor": "#f0f0f0",
        "borderRadius": 100,
        "borderWidth": 1,
        "padding": 10,
        "width": "50%",
      }
    }
  >
    <Text
      style={
        Object {
          "color": "white",
          "fontSize": 12,
          "textAlign": "center",
        }
      }
    >
      Load more
    </Text>
  </TouchableOpacity>
</View>
`;
